spi/winbond.c

```diff
@@ -285,6 +295,166 @@ static int winbond_is_locked(struct spi_flash *flash)
 	return 0;
 }
 
+static int winbond_sec_read(struct spi_flash *flash, u32 offset, size_t len, void *buf)
+{
+	int ret = 1;
+	u8 cmd[5];
+	u8 reg = (offset >> 8) & 0xFF;
+	u8 addr = offset & 0xFF;
+
+	if (reg != ADDR_W25_SEC1 && reg != ADDR_W25_SEC2 && reg != ADDR_W25_SEC3) {
+		spi_debug("SF: Wrong security register\n");
+		return 1;
+	}
+
+	cmd[0] = CMD_W25_RD_SEC;
+	cmd[1] = 0x0;
+	cmd[2] = reg;
+	cmd[3] = addr;
+	cmd[4] = 0x0; // dummy
```

Comment:
@mek-x I'm not getting what dummy in this context mean.

Reply:
AFAIC this is winbond fast read instruction. The instruction is as follows: 
DI line -> instrbyte-24bitaddr-dummybyte
DOline -------------------------------------------- Databyte1-Databyte2
For a time diagram @pietrushnic please refer to page 30: https://www.pjrc.com/store/w25q64fv.pdf

Reply:
yeah, this is dummy byte is in the datasheet and is required. The same is used in page write operation.